src/Components/MailtoDropdown.js

```diff
@@ -129,7 +128,9 @@ export default class MailtoDropdown extends Component {
                     if (!props.letter) e.preventDefault();
                     else {
                         if (MAILTO_HANDLERS[h].onClick)
-                            MAILTO_HANDLERS[h].onClick(data, props.createModal, props.onSuccess);
+                            MAILTO_HANDLERS[h].onClick(data, props.createModal).then(() => {
+                                props.onSuccess();
+                            });
```

Comment:
```suggestion
                            MAILTO_HANDLERS[h].onClick(data, props.createModal).then(props.onSuccess());
```

Reply:
Isn't that wrong? I think you mean this:

```suggestion
                            MAILTO_HANDLERS[h].onClick(data, props.createModal).then(props.onSuccess);
```